src/routes/AddDoctor.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { doctorsCreateThunk } from '../redux/doctor/doctorSlice';
+import { getUser } from '../redux/auth/authSlice';
+
+const AddDoctor = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const fileInputRef = useRef(null);
+
+  const [name, setName] = useState('');
+  const [bio, setBio] = useState('');
+  const [rate, setRate] = useState('');
+  const [specialization, setSpecialization] = useState('');
+  const [image, setImage] = useState('');
+  const [hospital, setHospital] = useState('');
+
+  const user = useSelector(getUser);
+
+  useEffect(() => {
+    if (!user) navigate('/auth');
```

Comment:
This logic is duplicated here. It is not necessary because the same is already written in `Layout.jsx`